### DISCUSSION TITLE
Can I use conductor-ui only as a viewer without authentication?

### DISCUSSION THREAD

**Discussion by earlbread** [Netflix/conductor#2445] [Q&A]
Hello, I'm considering using conductor, and I expect there will be very few use cases to manipulate the workflow in the UI. Is it possible to use the UI only in viewer mode without additional authentication or authorization?

**Comment by flavioschuindt** [comment]
By default the UI has no authentication/authorization at all. It is just a simple UI application. You can just make it up and running and will just work. In the future, if you want to add any auth/authZ you can put it behind an API Gateway which can take care of it for you.

**Reply by earlbread** [reply]
I know there is no authentication by default. What I am curious about is whether it can be used only in viewer mode without authentication. I mean, I want to remove all write functions.

**Reply by aravindanr** [reply]
Can you elaborate what you are referring to as a "write" operation? The UI does not allow creation of workflows, tasks or event handlers. Those are possible only using the apis.

**Reply by earlbread** [reply]
By default, we can restart workflow or retry task. But we can restrict the operations in viewer role with authentication. I wonder if it can be used only as a viewer without authentication.

**Reply by aravindanr** [reply]
In the current ui, it's not possible. The [new UI coming in 3.3.0](https://github.com/Netflix/conductor/discussions/2461), might have better ways to accomplish this. cc: @peterlau

**Reply by earlbread** [reply]
Got it, thank you for your answers.

**Reply by peterlau** [reply]
In the new 3.x UI authentication is completely decoupled from the UI implementation. However as stateful actions are restricted to buttons in a few places it would not be difficult to implement the effect you desire.  It would look something like:
1. Implement `useAuthentication` to retrieve session state. Precise implementation would depend on your auth strategy.
2. Wrap those action buttons in a predicate to not display unless authN & authZ context matches the desired values.
